code/game/machinery/omnibox.dm

```diff
@@ -0,0 +1,264 @@
+	icon_state = "Vehicle Factory"
+	var/finished = 60
+	var/progress = 0
+	var/reset = 1800
+	var/iterations = 0
+	var/record = 0
+	var/efficiency = 0.2
+	var/ideal
+	var/phase1
+	var/phase2
+	var/phase3
+
+
+/obj/machinery/power/omnibox/factory/Initialize()
+	. = ..()
+	phase1 = rand(10)
+	phase2 = rand(10,15)
+	phase3 = rand(15,20)
+	ideal = phase1
+	desc += " The factory's instruments are currently harmonized to a [ideal] second wavelength."
+
+/obj/machinery/power/omnibox/factory/interact(mob/user)
+	if (!anchored)
+		to_chat(user,"<span class='warning'>This device must be anchored by a wrench!</span>")
+		return
+	if(!Adjacent(user) && (!isAI(user)))
+		return
+	user.set_machine(src)
+	var/list/dat = list()
+	dat += ("<b>[name]</b><br>")
+	if (active)
+		dat += ("Factory: <A href='?src=\ref[src];action=disable'>On</A><br>")
+	else
+		dat += ("Factory: <A href='?src=\ref[src];action=enable'>Off</A><br>")
+	dat += ("The current vehicle is [100*(progress/finished)]% complete. It will require [finished-progress] ideal cubes to finish the current vehicles.<br><br>")
+	if(record)
+		dat += ("<b>The last omnicube was processed with a deviation of [record] seconds relative to the ideal - as a result the omnicube contributed [efficiency*100]% of its full potential.</b><br>")
+	else
+		dat += ("<b>Current instruments are synchronized to an omnicube delay of [ideal] seconds.</b><br>")
+	dat += ("<br><A href='?src=\ref[src];action=close'>Close</A>")
+	var/datum/browser/popup = new(user, "vending", "Vehicle Factory", 400, 350)
+	popup.set_content(dat.Join())
+	popup.open()
+
+/obj/machinery/power/omnibox/factory/process()
+	if(active && !crit_fail && anchored)
+		for(var/obj/item/omnicube/O in orange(1, src))
+			qdel(O)
+			record = 0.1 * ((ideal*10)-(world.time-O.lifespan))
+			var/dev = abs(record)
+			if(dev < 4)
+				efficiency = (1-(0.25*round(dev)))
+			else
+				efficiency = 0.2
+			progress += efficiency
+			playsound(src, 'sound/machines/ding.ogg', 25, 0)
+			if(progress >= finished)
+				progress -= finished
+				new /obj/vehicle/space/speedbike/atmos(locate(min(x+1,world.maxx),y,z))
+				new /obj/vehicle/space/speedbike/repair(locate(x,min(y+1,world.maxy),z))
+				iterations = 0
+				ideal = phase1
+				return
+			if(round(progress/(finished/3)) > iterations)
+				iterations++
+				switch(iterations)
+					if(0)
+						ideal = phase1
+					if(1)
+						ideal = phase2
+					if(2)
+						ideal = phase3
+				playsound(src, 'sound/machines/warning-buzzer.ogg', 50, 0)
+				say("Prototype construction has entered Phase [iterations+1]. Instruments have been synchronized to a [ideal] second delay.")
+	else
+		active = FALSE
+		icon_state = "[src]0"
+
+obj/item/omnicube
+	name = "Omnicube"
+	desc = "A volatile energy container designed for use in the vehicle factory."
+	icon = 'icons/obj/machines/omnibox.dmi'
+	icon_state = "omnicube"
+	var/lifespan = 0
+
+obj/item/omnicube/Initialize()
+	. = ..()
+	lifespan = world.time
```

Comment:
what is this suppose to even do?

Reply:
Records the "birth time" of the object so that its "age/lifespan" can be calculated when its processed.